src/hybrid_reader.c

```diff
@@ -424,7 +424,8 @@ IndexIterator *NewHybridVectorIterator(HybridIteratorParams hParams) {
   hi->timeoutCtx = (TimeoutCtx){ .timeout = hParams.timeout, .counter = 0 };
   hi->runtimeParams.timeoutCtx = &hi->timeoutCtx;
 
-  if (hParams.childIt == NULL) {
+  if (hParams.childIt == NULL || hParams.query.k == 0) {
```

Comment:
We actually can't, because today we have to evaluate all the iterators because they might yield a field that is needed  somewhere else (we want to return 0 results and not cause a syntax error)